### DISCUSSION TITLE
TSC Meeting, January

### DISCUSSION THREAD

**Discussion by dbaluta** [thesofproject/sof#7380] [Team Posts]
Date: 23 January 2020
Skype: https://meet.lync.com/nxp1-nxp/daniel.baluta/V66DF888
Participants: 
* Intel: Liam @lgirdwood, Lech @lbetlej, Pierre @plbossart. 
* Google: Dylan @dgreid and Curtis @cujomalainey. 
* NXP:Daniel @dbaluta.

General
======
#### 1) Opens, Roadmap, feature discussion.(Liam)

No opens.

Several release milestones for v1.6, v1.7 etc have been added to github and members have been requested to add new features alongside assigning a targeted release and assignee. Liam has already created and assigned many features related to general roadmap. Others to review milestones and realign features if needed.

Daniel asked about compressed offload for media codecs on roadmap. Pierre mentioned that driver APIs exist but will need plumbed into the SOF core and IPC created. **AR: Daniel** to create features for FW and driver for compressed offload support

Daniel also asked about deep buffering. Pierre mentioned that some timestamping updates are probably required to help support this but most of the infrastructure is already in place. **AR: Pierre** to add feature for missing deep buffering tasks in FW and driver.

#### 2) GSoC, Outreachy (Liam, Daniel)

Intel and NXP are submitting GSoC proposals, mostly centered around debug. 
https://wiki.linuxfoundation.org/gsoc/2020-gsoc-sound-open-firmware

Proposal submitted from NXP accepted at Outreachy diversity program
https://kernelnewbies.org/OutreachyIntro 

#### 3) Linaro Connect (Daniel)

NXP will participate at Linaro connect. Daniel will talk about [Status of SOF on I.MX8](https://sched.co/ZZEn) 
 
 Firmware
 ========
#### 1) 1.4.2 FW release. (all)

Now tagged and binaries built and signed for Intel platforms. Decision made to place FW ibinaries and ldc debug files into new repo alongside binary topologies.

#### 2) 1.5 FW release (all)

Original milestone is end of January, but pushing back one week for CNY vacation (as several validation team members are located in Shanghai).

#### 3) 1.6 FW release plans (all)

Members asked to assign items for v1.6. Timeline is aligned.

#### 4) 3rd party processing (Pierre, Daniel)

this topic covered two separate threads

a) support for compressed streams (details below)
b) support for branded PCM processing, such as Dolby, Waves, etc. We will need to remove the dependency on a component ID, which does not scale, and instead use UUIDs. Also we need to agree on either using the existing component API, or defining an 'SOF" API (which has the drawback of requiring 2 wrappers).

Driver
======

#### 1) Upstream status (Liam)
Redhat now integrating SOF FW, kernel and UCMs. Some issues being reported around UCM, Pulseaudio, DMICs and code signing due to differences between some laptops and development HW. 

Ask for SKL/KBL support from Dylan and Community. Stub secondary code loader is solution and will allow Google to use KBL. Liam has volunteered to do POC.

#### 2) ARM/Intel specific code integration opens (Liam, Daniel)

We are still running in circles on this. Issue started from Daniel trying to abstract Intel specific snd_soc_acpi_mach member of generic snd_sof_pdata structure. Pierre commented that we could remove 'ignore_machine' field from the common path and so we do not really need the abstraction. The problem now is that when removing ignore_machine filed some Intel tests are failing. We are still investigating this.
**AR**: Pierre asked @fredoh9 to have a look.

#### 3) Multi Client Support (Liam)

multi-client support is e.g. needed to create two cards using the same DSP. it can also be used to support debug clients (IPC flood) or probes.
This was pushed initially by Intel but should also be useful on other platforms

#### 4) Compressed streams (probes, compressed audio) (Pierre, Daniel)

The extractor for probes makes use of the ALSA compressed API, so we need to support this API in SOF. The work for probes being reviewed on alsa-devel adds the relevant code. To extend this work to standard codecs, we will need additional topology and plumbing work.

### 5) Resource management split between Linux/FW

Daniel described NXP DSP architecture shares IP programming flows between DSP FW and host SW. Liam mentioned this is the same as OMAP4 DSP and can used DPCM ASoC code to achieve the IP programming synchronisation between DSP and host.

 Tooling
 =======
 #### 1) Cadence free toolchain release. (Liam)

Cadence had previously created an free xcc compiler version for SOF on BYT in the very early days. **AR Intel** to ask the same for all Intel ISA configs.

#### 2) Probes support (Liam)

ABI now agreed, hopefully be merged for v1.5

#### 3) UI and tools (Pierre, Marcin)

 Marcin on vacation, down to present in February.

Misc
===
#### 1) Fuzzing (Curtis)

SOF has fuzzing interface, Google has fuzzer infra. Plan is to see if we can get fuzzing running as part of SOF-ci (using open source fuzzers) as well as part of CrOS-CI (using internal fuzzers [only once we have in tree building for SOF]). 

#### 2) Firmware distribution (all)

This topic has been chuntering along on the mailing list for some time now. Agreement to create a single repository for all SOF output binaries (FW, ldc and topologies). This repo will be initially manually created, but will be populated by scripts for future releases.

**Comment by ranj063** [comment]
Sorry I missed the meeting. Can I please request more than just a 3hr notice next time?

**Comment by dbaluta** [comment]
@ranj063 I'm sorry for this :(. Will add you for the next meetings.

**Comment by dbaluta** [comment]
@lgirdwood can you please share the meeting minutes with us. @plbossart what would be the best way to share the meeting minutes with everyone? We could add them to the documentation repo OR should we send an email to sound-open-firmware mailing list.

**Comment by lgirdwood** [comment]
@dbaluta I only have partial minutes due to the in call IT update performed on my Windows laptop and subsequent reboots. I will add inline to the discussion above for other to also update.

**Comment by dbaluta** [comment]
@lgirdwood added the bits under my name.

**Comment by cujomalainey** [comment]
Updated Fuzzing
